### PR TITLE
Refactor task submit

### PR DESCRIPTION
**Pull request by gaoyan1998**
## Verify this pull request

This pull request is code cleanup without any test coverage.

*(or)*

This pull request is already covered by existing tests, such as *(please describe tests)*.

(or)

This change added tests and can be verified as follows:

### CODE REVIEW

File: dinky-core/src/test/java/org/dinky/core/JobManagerTest.java [hunk 1]
```diff
@@ -45,29 +44,21 @@ public class JobManagerTest {
...
-                0,
-                null,
-                new HashMap<>(),
-                new HashMap<>());
+        JobConfig config = JobConfig.builder()
+                .type(GatewayType.YARN_SESSION.getLongValue())
+                .useResult(true)
+                .useChangeLog(true)
+                .useAutoCancel(true)
+                .useSession(true)
```
Comment (Zzm0809) [inline dinky-core/src/test/java/org/dinky/core/JobManagerTest.java]: useSession 这个应该是被干掉了 之前的 session catalog

File: dinky-admin/src/main/java/org/dinky/service/impl/TaskVersionServiceImpl.java [hunk 2]
```diff
@@ -41,4 +49,39 @@ public List<TaskVersion> getTaskVersionByTaskId(Integer taskId) {
...
+        taskVersion.setId(null);
+
+        if (Asserts.isNull(task.getVersionId())) {
+            // FIRST RELEASE, ADD NEW VERSION
+            taskVersion.setVersionId(1);
+            task.setVersionId(1);
+            save(taskVersion);
+        } else {
+            // Explain that there is a version, you need to determine whether it is an old version after fallback
+            TaskVersion version = versionMap.get(task.getVersionId());
```
Comment (Zzm0809) [inline dinky-admin/src/main/java/org/dinky/service/impl/TaskVersionServiceImpl.java:80]: 需要处理下 创建/更新时间 之前存在所有时间都沿用 task 表的创建时间

Comment (gaoyan1998) [reply dinky-admin/src/main/java/org/dinky/service/impl/TaskVersionServiceImpl.java:80]: done

**Review by Zzm0809 — APPROVED** [review]
LGTM

**Pull request merged**